src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -0,0 +1,213 @@
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.subsystems;
+
+import com.revrobotics.CANEncoder;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMax.IdleMode;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+
+import edu.wpi.first.wpilibj.SpeedControllerGroup;
+import edu.wpi.first.wpilibj.drive.DifferentialDrive;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants.DriveConstants;
+import edu.wpi.first.wpilibj.smartdashboard.SendableChooser;
+import edu.wpi.first.wpilibj.Solenoid;
+
+public class Drivetrain extends SubsystemBase {
+
+  //Creates variables for motor groups and differential drive
+
+  private final CANSparkMax m_leftMotor1;
+  private final CANSparkMax m_leftMotor2;
+  private final CANSparkMax m_rightMotor1;
+  private final CANSparkMax m_rightMotor2;
+
+  private final SpeedControllerGroup m_leftMotors;
+  private final SpeedControllerGroup m_rightMotors;
+
+  private final DifferentialDrive m_Drive;
+
+  //Creates variables for encoders
+
+  private final CANEncoder m_leftMotorEncoder1;
+  private final CANEncoder m_leftMotorEncoder2;
+  private final CANEncoder m_rightMotorEncoder1;
+  private final CANEncoder m_rightMotorEncoder2;
+
+  private boolean m_lowGear = false;
+
+  private String m_driveType = "Tank"; // adds a String object to store the drive type in
+  SendableChooser<String> m_chooser = new SendableChooser<>();
+
+  private final Solenoid gearShift;
+
+// Creates a new Drivetrain.
+public Drivetrain() {
+
+    // Sets variables equal to their ports and types
+    m_leftMotor1 = new CANSparkMax(DriveConstants.kLeftDriveMotor1Port, MotorType.kBrushless);
+    m_leftMotor2 = new CANSparkMax(DriveConstants.kLeftDriveMotor2Port, MotorType.kBrushless);
+    m_rightMotor1 = new CANSparkMax(DriveConstants.kRightDriveMotor1Port, MotorType.kBrushless);
+    m_rightMotor2 = new CANSparkMax(DriveConstants.kRightDriveMotor2Port, MotorType.kBrushless);
+
+    motorInit(m_rightMotor1, DriveConstants.kRightEncoderInverted); 
+    motorInit(m_rightMotor2, DriveConstants.kRightEncoderInverted);
+    motorInit(m_leftMotor1, DriveConstants.kLeftEncoderInverted);
+    motorInit(m_leftMotor2, DriveConstants.kLeftEncoderInverted); 
+    m_leftMotors = new SpeedControllerGroup(m_leftMotor1, m_leftMotor2);
+    m_rightMotors = new SpeedControllerGroup(m_rightMotor1, m_rightMotor2);
+
+    m_Drive = new DifferentialDrive(m_leftMotors, m_rightMotors);
+
+     // adds options to the drive type
+     m_chooser.addOption("Tank Drive", "Tank");
+     m_chooser.addOption("Arcade Drive", "Arcade");
+     m_chooser.addOption("Trigger Drive", "Trigger");
+
```

Comment:
After adding the options to the chooser, we also need to send it to the SmartDashboard.
```suggestion
SmartDashboard.putData("Drive Control", m_chooser);
```

Reply:
added it to smart dashboard